xbmc/network/websocket/WebSocketV13.cpp

```diff
@@ -102,7 +104,11 @@ bool CWebSocketV13::Handshake(const char* data, size_t length, std::string &resp
 
   // There must be a "Connection" header with the value "Upgrade"
   value = header.getValue(WS_HEADER_CONNECTION_LC);
-  if (value == NULL || strnicmp(value, WS_HEADER_UPGRADE, strlen(WS_HEADER_UPGRADE)) != 0)
+  if (value == nullptr)
+    elements = std::vector<std::string>();
+  else
+    elements = StringUtils::Split(value, ",");
+  if (elements.empty() || !std::any_of(elements.cbegin(), elements.cend(), [](std::string elem) { return StringUtils::EqualsNoCase(StringUtils::Trim(elem), WS_HEADER_UPGRADE); }))
```

Comment:
can you pass `elem` by reference or use move constructors?

Reply:
pass by reference doesn't work (StringUtils::Trim doesn't accept const)
`... StringUtils::Trim(std::move(elem)) ...` is the move syntax you mean?

Reply:
yup